modules/gateway/helpers.go

```diff
@@ -1,57 +1,60 @@
 package gateway
 
 import (
-	"errors"
 	"io/ioutil"
-	"math/rand"
-	"sync"
+	"net"
+	"net/http"
 
 	"github.com/NebulousLabs/Sia/encoding"
-	"github.com/NebulousLabs/Sia/network"
+	"github.com/NebulousLabs/Sia/modules"
 )
 
-func (g *Gateway) addPeer(peer network.Address) error {
-	if _, exists := g.peers[peer]; exists {
-		return errors.New("peer already added")
-	}
-	g.peers[peer] = struct{}{}
-	return nil
+var pong = [4]byte{'p', 'o', 'n', 'g'}
+
+// Ping returns whether an Address is reachable and responds correctly to the
+// ping request -- in other words, whether it is a potential peer.
```

Comment:
it would be pretty easy to insert magic bytes into all network communication if need be. Like <10 line diff.